### PR TITLE
using clang-format & test not support windows

### PR DESCRIPTION
**Pull request by jinfeihan57**

### CODE REVIEW

**Comment by kuba--** [comment]
take a look into macos build. it's falling because of no new line in zip.c file

**Comment by codylico** [comment]
(more outside comments, ignore if unwanted) I think I may know why `test_extract_stream` fails on Windows.
```c
  fseek(fp, 0L, SEEK_END);
  size_t filesize = ftell(fp);
  fseek(fp, 0L, SEEK_SET);
  // ...
  size_t size = fread(stream, sizeof(char), filesize, fp);
  assert(filesize == size);
```
This method of finding a file size easily breaks on Windows when reading in text mode. (Windows seeks based on byte count, but silently converts "\r\n" to "\n", decreasing actual bytes read.)

You could try to adjust the mode parameter of the `fopen` call to open in binary mode:
```c
// remove this:
  fp = fopen(ZIPNAME, "r+");
// ..., and use this instead:
  fp = fopen(ZIPNAME, "rb+");
```

**Comment by jinfeihan57** [comment]
ok

**Comment by jinfeihan57** [comment]
@kuba-- 
Any suggestions about mac failure.

**Comment by kuba--** [comment]
```
Users/runner/work/zip/zip/src/zip.c:1026:2: error: no newline at end of file [-Werror,-Wnewline-eof]
}
 ^
1 error generated.
make[2]: *** [CMakeFiles/OBJLIB.dir/src/zip.c.o] Error 1
make[1]: *** [CMakeFiles/OBJLIB.dir/all] Error 2
make: *** [all] Error 2
```

**Comment by jinfeihan57** [comment]
@kuba-- 
It's all done.

**Comment by jinfeihan57** [comment]
> > @kuba--
> > Did you mean clang-format ?
> > I did it there is nothing to commit.
> 
> Just when you're done run following script:
> 
> ```
> for file in $(git ls-files | \grep -E '\.(c|h)$' | \grep -v -- '#')
> do
>     clang-format -i $file
> done
> ```

I've already done it! nothing to commit, except clang-format.sh.

截图

**Comment by kuba--** [comment]
> > > @kuba--
> > > Did you mean clang-format ?
> > > I did it there is nothing to commit.
> > 
> > 
> > Just when you're done run following script:
> > ```
> > for file in $(git ls-files | \grep -E '\.(c|h)$' | \grep -v -- '#')
> > do
> >     clang-format -i $file
> > done
> > ```
> 
> I've already done it! nothing to commit, except clang-format.sh.
> 
> 截图

Ok, so maybe this time formatting didn't change.

File: src/zip.c [hunk 1]
```diff
@@ -1021,5 +1023,4 @@ int zip_extract_stream(const char *stream, size_t size, const char *dir,
   free(zip_archive);
 
   return status;
-
 }
```
Comment (kuba--) [inline src/zip.c]: we need a new line at the end of file  for macos builds

File: src/zip.c [hunk 2]
```diff
@@ -950,10 +935,92 @@ int zip_extract(const char *zipname, const char *dir,
...
   // Close the archive, freeing any resources it was using
-  if (!mz_zip_reader_end(&zip_archive)) {
+  if (!mz_zip_reader_end(zip_archive)) {
     // Cannot end zip reader
     status = -1;
   }
+  return status;
+}
+
+static inline mz_zip_archive *zip_archive_init_file(const char *zipname,
```
[the remark this answers is not in the document]
Comment (kuba--) [reply src/zip.c:945]: I think we don't need this function. 
It basically `mallocs` and call `mz_zip_reader_init_file`. So I think we can just pass reference to `zip_archive` to `mz_zip_reader_init_file` on higher level.

File: src/zip.c [hunk 3]
```diff
@@ -950,10 +935,92 @@ int zip_extract(const char *zipname, const char *dir,
...
+    free(zip_archive);
+    return NULL;
+  }
+  // Now try to open the archive.
+  if (!mz_zip_reader_init_mem(zip_archive, stream, size, flags)) {
+    // Cannot initialize zip_archive reader
+    free(zip_archive);
+    return NULL;
+  }
+  return zip_archive;
```
Comment (kuba--) [inline src/zip.c:1005]: The same here, I think we don't need this function. just pass reference to `zip_archive` and call `mz_zip_reader_init_mem`

**Comment by jinfeihan57** [comment]
It's all done. check again.

**Review by kuba-- — APPROVED** [review]

**Pull request merged**